### PR TITLE
Clean up collection entity

### PR DESCRIPTION
**Pull request by kulyk**
Cleans up collection entity code a little bit:

* moves entity-related tests from `frontend/test/metabase/entities/collections/*` to `frontend/src/metabase/entities/collections/tests`
* extracts big selectors like `getExpandedCollectionsById` and `getInitialCollection` to dedicated files
* converts whatever was straightforward to TypeScript
* simplifies bits of code here and there

Based on #26477

### CODE REVIEW

File: frontend/src/metabase/nav/containers/MainNavbar/MainNavbarContainer/MainNavbarContainer.tsx [hunk 1]
```diff
@@ -105,6 +100,7 @@ function MainNavbarContainer({
     );
 
     preparedCollections.push(...userPersonalCollections);
+    ``;
```
Comment (alxnddr) [inline frontend/src/metabase/nav/containers/MainNavbar/MainNavbarContainer/MainNavbarContainer.tsx:103]: typo?

Comment (kulyk) [reply frontend/src/metabase/nav/containers/MainNavbar/MainNavbarContainer/MainNavbarContainer.tsx:103]: Yup! Thank you

File: frontend/src/metabase/nav/containers/MainNavbar/MainNavbarContainer/MainNavbarView.tsx [hunk 2]
```diff
@@ -188,7 +188,7 @@ function CollectionSectionHeading({
         </SidebarLink>
         {currentUser.is_superuser && (
           <SidebarLink
-            icon={getCollectionIcon(PERSONAL_COLLECTIONS)}
+            icon={getCollectionIcon(PERSONAL_COLLECTIONS as Collection)}
```
Comment (alxnddr) [inline frontend/src/metabase/nav/containers/MainNavbar/MainNavbarContainer/MainNavbarView.tsx:191]: The `Collection` type does not allow `id=personal` but the `PERSONAL_COLLECTIONS` has such id. Wdyt about converting `frontend/src/metabase/entities/collections/constants.js` to TS?

**Review by alxnddr — APPROVED** [review]

**Review by iethree — APPROVED** [review]
🏆

File: frontend/src/metabase/nav/containers/MainNavbar/MainNavbarContainer/MainNavbarView.tsx (hunk 2 above)
Comment (kulyk) [reply frontend/src/metabase/nav/containers/MainNavbar/MainNavbarContainer/MainNavbarView.tsx:191]: Good call. I've tried that and ran into exactly this issue. I'll see if I can put together another PR for this. We'd most likely have to change how we work with these "hardcoded" collections

**Pull request merged**